File: mifi_rental/android/app/src/main/kotlin/com/ucloudlink/mifi_rental/pay/JSApi.kt
package com.ucloudlink.mifi_rental.pay

import android.app.Activity
import android.webkit.JavascriptInterface
import com.ucloudlink.core_log.ULog
import com.ucloudlink.mifi_rental.Router

class JSApi(private val params: String?, private val activity: PayActivity) {
    /***
     * 同步API.
     * public any handler(Object msg)
     * 参数必须是 Object 类型，并且必须申明（如果不需要参数，申明后不适用即可）。返回值类型没有限制，可以是任意类型。
     * 异步 API.
     * public void handler(Object arg, CompletionHandler handler)
     */
    @JavascriptInterface
    fun getInfo(msg: Any): String? {
        ULog.d("JSApi getInfo $msg params:$params")
        return params
    }

    @JavascriptInterface
    fun jumpToNativePage(msg: Any): String? {
        ULog.i("JSApi jumpToNativePage $msg")
        when (msg) {
            "toPaySuccess" -> {
//                val exts = HashMap<String, Any>()
//                exts["clearTask"] = true
//                Router.openPageByUrl(activity, "flutter://query", null, exts = exts)
                activity.paySuccess()
            }
            "toPayCancel", "repayAgain" -> {
                activity.cancelOrder()
            }
            "toContactUsActivity" -> {
            }

        }
        return ""
    }
}

File: mifi_rental/ios/Runner/JSApi.swift
//
//  JSApi.swift
//  Runner
//
//  Created by yyg on 2020/5/14.
//  Copyright © 2020 The Chromium Authors. All rights reserved.
//

import Foundation
import SwiftyJSON

typealias JSCallback = (String, Bool) -> Void

class JSApi: NSObject {

    var params: Dictionary<String , Any>? = nil
    weak var viewController: UINavigationController?
    
    init(params: Dictionary<String , Any>?, viewController: UINavigationController?) {
        self.params = params
        self.viewController = viewController
    }
    
    @objc func getInfo(_ arg: String) -> String? {
        self.params?.removeValue(forKey: "__callback_id__")
        self.params?.removeValue(forKey: "terminalSn")
        self.params?["langType"] = "zh-CN"
        self.params?["localeCode"] = "zh_CN"
        return JSON(self.params as Any).rawString()
    }
    
    @objc func jumpToNativePage(_ arg: String) -> String? {
        switch arg {
        case "toPaySuccess":
            // 支付完成到设备查询界面
            self.viewController?.dismiss(animated: true, completion: nil)
            NotificationCenter.default.post(name: NSNotification.Name(rawValue: "paySuccess"), object: nil)
//            PlatformRouterImp.sharedInstance.open("query", urlParams: ["":""], exts: ["animated" : true]) { _ in }
            break
        case "toPayCancel", "repayAgain":
            // 关闭支付界面并且取消订单
            // NotificationCenter.default.post(name: NSNotification.Name(rawValue: "cancelOrder"), object: nil)
            self.viewController?.dismiss(animated: true, completion: nil)
            break
        case "toContactUsActivity":
            break
        default:
            break
        }
        return ""
    }
}
